Block/BaseHyvaGrid.php

```diff
@@ -4,30 +4,54 @@
 
 use Hyva\Admin\ViewModel\HyvaGridInterface;
 use Hyva\Admin\ViewModel\HyvaGridInterfaceFactory;
+use Magento\Framework\View\Element\BlockInterface;
 use Magento\Framework\View\Element\Template;
 
 abstract class BaseHyvaGrid extends Template
 {
     private HyvaGridInterfaceFactory $gridFactory;
+    /**
+     * @var array
+     */
+    private $children;
+
+    private $grid;
```

Comment:
In that case I made this because a block that is instantiated with the grid_name can keep the grid instantiated. In my opinion there is no need to always create a new grid when calling getGrid(). The GridBlock knows about the grid and this should probably not change, right?